src/v/ssx/sformat.h

```diff
@@ -24,6 +24,18 @@ struct fmt::formatter<seastar::sstring> {
     }
 };
 
+#if FMT_VERSION >= 80000
```

Comment:
not sure if `ssx/sformat.h` is the best place for this wrapper. i assume "ssx" is short for seastar extension, but `fmt_append()` is but a wrapper of `fmt::format_to()`, which is not directly connected with Seastar..

Reply:
>  i assume "ssx" is short for seastar extension

yes that's right.

> not sure if ssx/sformat.h is the best place for this wrapper.

I'll let Ben chime in here, but I think you are right. My guess is that these are only used by `ssx::sformat` (specialization for `seastar::sstring`) and if we had wider usage in the tree we'd move this to a more generic location.

Reply:
> > not sure if ssx/sformat.h is the best place for this wrapper.
> 
> I'll let Ben chime in here, but I think you are right. My guess is that these are only used by `ssx::sformat` (specialization for `seastar::sstring`) and if we had wider usage in the tree we'd move this to a more generic location.

i am afraid this is not quite right. i am trying to address the FTBFS caused by `fmt::format_to(format_str, ...)` where `format_str` is not constexpr. and we do have a handful of callers which simply pass the `format_str`to the underlying `fmt::format_to()`. they are using `fmt::format_to()` directly. even the `format_str` is a constexpr when it gets passed to these callers. but the `constexpr`ness is erased when it is passed down to `fmt::format_to()`, hence the error.

Reply:
i will take a look at the changelog/git history of {fmt} to find out the minimum version where `fmt::format_to(std::back_inserter(buf), format_str, std::forward<Args>(args)...);` works in hope to drop the deprecated version. but i am not quite optimistic in this regard.. as Seastar claims to support fmt 5.0.0. see https://github.com/scylladb/seastar/blob/acf7e3523ba7e20d1bfb09c92a705dcc3e7f0161/cmake/SeastarDependencies.cmake#L87 .

Reply:
Thanks Kefu! @BenPope could take a look? This is probably something we want to address now or right after we cut our next release and bump clang and seastar and other dependency versions?